test/functional/ES512TokenTest.php

```diff
@@ -0,0 +1,280 @@
+            ->permittedFor('http://client.abc.com')
+            ->issuedBy('http://api.abc.com')
+            ->withClaim('user', ['name' => 'testing', 'email' => 'testing@abc.com'])
+            ->getToken($this->config->getSigner(), new Key('testing'));
+    }
+
+    /**
+     * @test
+     *
+     * @covers \Lcobucci\JWT\Configuration
+     * @covers \Lcobucci\JWT\Token\Builder
+     * @covers \Lcobucci\JWT\Token\Plain
+     * @covers \Lcobucci\JWT\Token\DataSet
+     * @covers \Lcobucci\JWT\Token\Signature
+     * @covers \Lcobucci\JWT\Signer\Key
+     * @covers \Lcobucci\JWT\Signer\Ecdsa
+     * @covers \Lcobucci\JWT\Signer\Ecdsa\MultibyteStringConverter
+     * @covers \Lcobucci\JWT\Signer\Ecdsa\Sha256
+     * @covers \Lcobucci\JWT\Signer\OpenSSL
+     */
+    public function builderShouldRaiseExceptionWhenKeyIsNotEcdsaCompatible(): void
+    {
+        $builder = $this->config->createBuilder();
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('This key is not compatible with this signer');
+
+        $builder->identifiedBy('1')
+            ->permittedFor('http://client.abc.com')
+            ->issuedBy('http://api.abc.com')
+            ->withClaim('user', ['name' => 'testing', 'email' => 'testing@abc.com'])
+            ->getToken($this->config->getSigner(), static::$rsaKeys['private']);
+    }
+
+    /**
+     * @test
+     *
+     * @covers \Lcobucci\JWT\Configuration
+     * @covers \Lcobucci\JWT\Token\Builder
+     * @covers \Lcobucci\JWT\Token\Plain
+     * @covers \Lcobucci\JWT\Token\DataSet
+     * @covers \Lcobucci\JWT\Token\Signature
+     * @covers \Lcobucci\JWT\Signer\Key
+     * @covers \Lcobucci\JWT\Signer\Ecdsa
+     * @covers \Lcobucci\JWT\Signer\Ecdsa\MultibyteStringConverter
+     * @covers \Lcobucci\JWT\Signer\Ecdsa\Sha512
+     * @covers \Lcobucci\JWT\Signer\OpenSSL
+     */
+    public function builderCanGenerateAToken(): Token
+    {
+        $user    = ['name' => 'testing', 'email' => 'testing@abc.com'];
+        $builder = $this->config->createBuilder();
+
+        $token = $builder->identifiedBy('1')
+            ->permittedFor('http://client.abc.com')
+            ->permittedFor('http://client2.abc.com')
+            ->issuedBy('http://api.abc.com')
+            ->withClaim('user', $user)
+            ->withHeader('jki', '1234')
+            ->getToken($this->config->getSigner(), $this->config->getSigningKey());
+
+        self::assertEquals('1234', $token->headers()->get('jki'));
+        self::assertEquals('http://api.abc.com', $token->claims()->get(Token\RegisteredClaims::ISSUER));
+        self::assertEquals($user, $token->claims()->get('user'));
+
+        self::assertEquals(
+            ['http://client.abc.com', 'http://client2.abc.com'],
+            $token->claims()->get(Token\RegisteredClaims::AUDIENCE)
+        );
+
+        return $token;
+    }
+
+    /**
+     * @test
+     * @depends builderCanGenerateAToken
```

Comment:
`@depends` can likely be skipped overall

Reply:
Not really since it depends on the value returned by it :)